File: DSA/SakshamKashyap_590014183/day75/day75_Question_1.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def print_ancestors(root, target):
    if not root:
        return False
    if root.val == target:
        return True
    if print_ancestors(root.left, target) or print_ancestors(root.right, target):
        print(root.val, end=' ')
        return True
    return False

# Example usage:
if __name__ == "__main__":
    # Tree:    1
    #         / \
    #        2   3
    #       / \
    #      4   5
    root = TreeNode(1, TreeNode(2, TreeNode(4), TreeNode(5)), TreeNode(3))
    print_ancestors(root, 5)  # Output: 2 1
    print()

File: DSA/VenikaPanwar_590013755/Day_75_Question_1.java
import java.util.*;

public class Day_75_Question_1 {

    static class TreeNode {
        int val;
        TreeNode left, right;
        TreeNode(int val) { this.val = val; }
    }

    // Prints the left view of the binary tree
    public static void printLeftView(TreeNode root) {
        if (root == null) return;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while (!queue.isEmpty()) {
            int levelSize = queue.size();
            for (int i = 0; i < levelSize; i++) {
                TreeNode node = queue.poll();
                if (i == 0) System.out.print(node.val + " "); // First node at this level
                if (node.left != null) queue.offer(node.left);
                if (node.right != null) queue.offer(node.right);
            }
        }
        System.out.println();
    }

    public static void main(String[] args) {
        // Example: Tree 1 / \ 2 3 \ 4
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.right = new TreeNode(4);

        System.out.print("Left View: ");
        printLeftView(root); // Output: 1 2 4
    }
}